### ISSUE TITLE
AttributeError: 'str' object has no attribute 'decode'

### ISSUE THREAD

**Issue by lexxxel** [eoyilmaz/displaycal-py3#68]
Hi everyone,

first of all, thank you so much for taking the time migrate DisplayCal to python3 :heart: 

I run a calibration with my Spider5 and got an issue after the calibration was complete:

```
DisplayCAL: Starting interaction with subprocess


Number of patches = 175

Setting up the instrument

Instrument Type:   Datacolor Spyder5

Serial Number:     50237613

Hardware version:  0x0a0f

argyll-dispread: Warning - new_dispwin: Expected VideoLUT depth 8 doesn't match actual 10

Place instrument on test window.

DisplayCAL: Detected instrument placement (screen/spot) message
DisplayCAL: Waiting for send buffer
DisplayCAL: Skipping place instrument on screen message...
DisplayCAL: Send buffer received:  
DisplayCAL: Sending buffer: ' '
Hit Esc or Q to give up, any other key to continue:

Patch 175 of 175

The instrument can be removed from the screen.

Written 'DisplayPort-0 #1 2022-03-24 17-34 130cdm² D6500 2.2 F XYZLUT+MTX.ti3'

DisplayCAL: Reached EOF (OK)
argyll-dispread exitcode: 0
value could not be a str: 1931_2
Create profile from measurement data...
Preparing ChromaticityType tag from TI3 colorants
Storing settings in TI3
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/wx/lib/delayedresult.py", line 267, in wrapper
    result = workerFn(*args, **kwargs)
  File "/home/apit/projects/displaycal-py3/DisplayCAL/worker.py", line 1839, in __call__
    result = self.producer(*args, **kwargs)
  File "/home/apit/projects/displaycal-py3/DisplayCAL/worker.py", line 10623, in create_profile
    cmd, args = self.prepare_colprof(
  File "/home/apit/projects/displaycal-py3/DisplayCAL/worker.py", line 13356, in prepare_colprof
    ti3[0].fix_zero_measurements(logfile=self.get_logfiles(False))
  File "/home/apit/projects/displaycal-py3/DisplayCAL/CGATS.py", line 2030, in fix_zero_measurements
    logfile.write(
  File "/home/apit/projects/displaycal-py3/DisplayCAL/util_io.py", line 180, in write
    data = data.decode()
AttributeError: 'str' object has no attribute 'decode'
```

with the issue
```python
    def write(self, data):
        data = data.decode() <---- here
        data = data.replace(self.linesep_in, "\n")
        for char in data:
            if char == "\r":
                while self.buf and not self.buf.endswith(self.linesep_out):
                    self.buf = self.buf[:-1]
            else:
                if char == "\n":
                    self.buf += self.linesep_out
                    self.commit()
                else:
                    self.buf += char
```

with `data`:
![image](https://user-images.githubusercontent.com/5159138/159999453-fdc80c9f-267d-4c7f-869b-2a37d1b73a53.png)

**Comment by lexxxel** [comment]
I changed the code to:
```diff
diff --git a/DisplayCAL/util_io.py b/DisplayCAL/util_io.py
index 588e22a4..651c9fd1 100644
--- a/DisplayCAL/util_io.py
+++ b/DisplayCAL/util_io.py
@@ -177,7 +177,8 @@ class LineBufferedStream(object):
             self.buf = ""
 
     def write(self, data):
-        data = data.decode()
+        if type(data) is not str:
+            data = data.decode()
         data = data.replace(self.linesep_in, "\n")
         for char in data:
             if char == "\r":
```
to get it working. Not elegant ... but at least it works :-/

**Comment by eoyilmaz** [comment]
Thanks for the issue report it is fixed in 1265482
